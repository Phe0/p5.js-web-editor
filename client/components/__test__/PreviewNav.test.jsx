import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PreviewNav from '../PreviewNav';

describe('PreviewNav', () => {
  let wrapper = null;
  const props = {
    owner: {
      username: 'testing-owner'
    },
    project: {
      id: '1',
      name: 'testing-project',
    }
  };

  beforeEach(() => {
    wrapper = shallow(<PreviewNav {...props} />);
  });

  it('it renders preview-nav', () => {
    expect(wrapper.exists('.preview-nav')).toEqual(true);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<PreviewNav {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
