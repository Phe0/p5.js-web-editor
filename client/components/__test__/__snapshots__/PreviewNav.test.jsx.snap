// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewNav renders correctly 1`] = `
<nav
  className="nav preview-nav"
>
  <div
    className="nav__items-left"
  >
    <div
      className="nav__item-logo"
    >
      <test-file-stub
        aria-label="p5.js Logo"
        className="svg__logo"
        focusable="false"
        role="img"
      />
    </div>
    <a
      className="nav__item"
      onClick={[Function]}
      style={Object {}}
    >
      testing-project
    </a>
    <p
      className="toolbar__project-owner"
    >
      by
    </p>
    <a
      className="nav__item"
      onClick={[Function]}
      style={Object {}}
    >
      testing-owner
    </a>
  </div>
  <div
    className="nav__items-right"
  >
    <a
      aria-label="Edit Sketch"
      onClick={[Function]}
      style={Object {}}
    >
      <test-file-stub
        aria-hidden="true"
        className="preview-nav__editor-svg"
        focusable="false"
      />
    </a>
  </div>
</nav>
`;
